### PR TITLE
Update MSAL to reflect proposed OM - Request Component.

### PR DESCRIPTION
**Pull request by kpanwar**
This issue updates the source to move from handlers to Request model. #97

### CODE REVIEW

File: src/MSAL.PCL/AbstractClientApplication.cs [hunk 1]
```diff
@@ -198,7 +198,7 @@ protected AbstractClientApplication(string authority, string clientId, string re
                 parameters = PlatformPlugin.DefaultPlatformParameters;
             }
 
-            var handler = new AcquireTokenSilentHandler(this.GetHandlerData(authenticator, scope, policy, this.UserTokenCache), userIdentifier,  parameters, forceRefresh);
+            var handler = new SilentRequest(this.GetHandlerData(authenticator, scope, policy, this.UserTokenCache), userIdentifier,  parameters, forceRefresh);
```
Comment (weijjia) [inline src/MSAL.PCL/AbstractClientApplication.cs]: whooooo!! No more handlers. :)

File: src/MSAL.PCL/AbstractClientApplication.cs [hunk 2]
```diff
@@ -209,13 +209,13 @@ protected AbstractClientApplication(string authority, string clientId, string re
                 parameters = PlatformPlugin.DefaultPlatformParameters;
             }
 
-            var handler = new AcquireTokenSilentHandler(this.GetHandlerData(authenticator, scope, policy, this.UserTokenCache), user, parameters, forceRefresh);
+            var handler = new SilentRequest(this.GetHandlerData(authenticator, scope, policy, this.UserTokenCache), user, parameters, forceRefresh);
             return await handler.RunAsync().ConfigureAwait(false);
         }
 
-        internal virtual HandlerData GetHandlerData(Authenticator authenticator, string[] scope, string policy, TokenCache cache)
+        internal virtual RequestData GetHandlerData(Authenticator authenticator, string[] scope, string policy, TokenCache cache)
```
Comment (weijjia) [inline src/MSAL.PCL/AbstractClientApplication.cs]: Check the latest architecture doc, I think we change it to AuthenticationRequestParameters

Comment (kpanwar) [inline src/MSAL.PCL/AbstractClientApplication.cs]: fixed.

**Comment by weijjia** [comment]
:shipit:  :sheep:

**Pull request merged**